autoendpoint/src/extractors/notification.rs

```diff
@@ -160,25 +161,28 @@ impl Notification {
     /// fields are still required when delivering to the connection server, so
     /// we can't simply convert this notification type to that one and serialize
     /// via serde.
-    pub fn serialize_for_delivery(&self) -> HashMap<&'static str, serde_json::Value> {
+    pub fn serialize_for_delivery(&self) -> ApiResult<HashMap<&'static str, serde_json::Value>> {
         let mut map = HashMap::new();
 
         map.insert(
             "channelID",
-            serde_json::to_value(self.subscription.channel_id).unwrap(),
+            serde_json::to_value(self.subscription.channel_id)?,
```

Comment:
I might be overthinking this but: usually killing unwraps is an improvement but with this moving the unwrap to the caller -- IMHO the original unwraps here were clearer that the individual serializations were infallible (probably because the line doing the unwrap hides the method details, somehow I don't even trust it even w/ the name `serialize_for_delivery`). Maybe renaming it to `serialize_for_delivery_infallible` would improve this, or better yet (IMHO) continue having the method not return a `Result` but without the unwraps w/ [what I did in this branch](https://github.com/mozilla-services/autopush-rs/compare/feat/SYNC-4324_track_db...feat/SYNC-4324_track_db-document-infallible)

Reply:
Maybe I'm just being dense. 
`serialize_for_delivery()` gets unwrapped by `send_notification()` because `send_notification()` wants to return a `reqwest::Error` that gets checked by `route_notification` to see which error metric needs to be updated.

That doesn't seem super clear either. Plus, the function isn't infallible, since it very much returns an error that we pay attention to. 

`route_notification()` is part of the `Router` impl, which returns an `ApiResult`. Isn't the problem that we're basically eating the error?  Wouldn't it be better to add `ApiErrorKind::ReqwestError(_)` and then handle the error without doing any unwraps?

Going to update with that as a patch.

Reply:
I think `serialize_for_delivery` is still infallible but yea, that definitely works (and I probably got too lost in the weeds)